internal/security/secretstoreclient/vault.go

```diff
@@ -171,21 +172,22 @@ func (vc *vaultClient) commonRequest(token *string, method string, path string,
 // delegate is a callback function called to handle the decoded response
 func (vc *vaultClient) processResponse(resp *http.Response, responseError error,
 	operationDescription string, expectedStatusCode int,
-	responseObject interface{}) (statusCode int, err error) {
+	responseObject interface{}) (int, error) {
 
 	if responseError != nil {
-		vc.logger.Error(fmt.Sprintf("unable to make request to %s failed: %s", operationDescription, err.Error()))
-		return 0, err
+		vc.logger.Error(fmt.Sprintf("unable to make request to %s failed: %s", operationDescription, responseError.Error()))
```

Comment:
IMO, logging should leave it to the caller if possible. so I would suggest just removing this logging here and just return the error.

Reply:
As processResponse() was a refactoring to reduce code duplication in the original code, I am going to leave as-is unless someone seconds it, because the error message adds critical context (i.e. which api call failed) that would be lost of we removed it.

Reply:
Well, this particular one does not seem to add any value of it because the responseError just passes through and the caller has everything in the message.  As a matter of the fact, the caller should have more knowledge on the which api called failed in this case since the response is passed in from the caller side.

Reply:
fwiw I agree with @jim-wang-intel here, but not necessary for fixup now

Reply:
@anonymouse64  I can further improve the refactor, but need some feedback.  Right now my hangup is that I have two different request() functions, depending on whether the request body has a json object, no request body at all, or potentially an arbitrary payload.  I can combine the request and response functions into a unified request workflow,  but Vault doesn't have named arguments.  I could:
(1) pass interface{} and do a type assertion on it. to determine what to do.  If it is nil or an io.Reader I can just send it on as the request body else assume it is a struct and send it through the JSON marshaller before sending it, or
(2) make a function that just takes a struct{} and have a member JSONData of type interface{} (if not nil, Marshal it) and a member RequestBodyReader of type io.Reader (pass this to the request instead of JSONdata is nil), or
(3) do the same as 2 above, but define the struct and create a method like Do() as a method on the struct.
I kinda lean towards 2 or 3 because 1 will result in a function with 10 or so positional arguments which is almost unreadable.

Reply:
I think this fixup is low priority and can wait until you've finished other higher priority tasks for Fuji, but FWIW I like 2